design/src/rfcs/rfc0012_unsynchronized_versioning.md

```diff
@@ -0,0 +1,207 @@
+handle this process: the previously released version number, the [smithy-rs] version used to generate the code,
+and the level of model updates being applied. For this last one, if there are multiple model updates that
+affect only documentation, but then one model update that affects an API, then as a whole they will be
+considered as affecting an API and require a `minor` version bump.
+
+The previously released version number will be retrieved from crates.io using its API. The [smithy-rs] version
+used during code generation will become a build artifact that is saved to `versions.toml` in [aws-sdk-rust].
+During phase 1, the tooling required to know if a model is a documentation-only change will not be available,
+so all model changes will result in a `minor` version bump during this phase.
+
+Overall, determining a generated crate's version number looks as follows:
+
+![Phase 1: How to version a generated crate](rfc0012_unsynchronized_versioning/phase1_generated_crate_version.svg)
+
+- __A: smithy-rs changed?__: Compare the `smithy_rs_version` in the previous `versions.toml` with the
+  next `versions.toml` file, and if the values are different, consider [smithy-rs] to have changed.
+- __B: model changed?__: Similarly, compare the `model_hash` for the crate in `versions.toml`.
+
+### Versioning for Runtime Crates
+
+The old scheme of all runtime crates in [smithy-rs] having a fake `0.0.0-smithy-rs-head` version number with
+a build step to replace those with a consistent `major.minor` will be removed. These runtime crates will begin
+having their actual next version number in the Cargo.toml file in smithy-rs.
+
+This introduces a new problem where a developer can forget to bump a runtime crate version, so a method of
+process enforcement needs to be introduced. This will be done through CI when merging into `smithy-rs/main`
+and repeated when merging into `aws-sdk-rust/main`.
+
+The following checks need to be run for runtime crates:
+
+![Phase 1: How to validate a runtime version bump](rfc0012_unsynchronized_versioning/phase1_runtime_crate_version_checks.svg)
+
+- __A: Crate has changed?__ The crate's source files and manifest will be hashed for the previous version
+  and the next version. If these hashes match, then the crate is considered unchanged.
+- __B: Minor bumped?__ The previous version is compared against the next version to see if the minor version
+  number was bumped.
+- __C: Version changed?__ The previous version is compared against the next version to see if it changed.
+- __D: Patch bumped?__ The previous version is compared against the next version to see if the patch version
+  number was bumped.
+- __E: Semverver passes?__ Runs [rust-semverver] against the old and new versions of the crate.
+  - If semverver fails to run (for example, if it needs to be updated to the latest nightly to succeed),
+    then fail CI saying that either semverver needs to be updated, or that a minor version bump is required.
```

Comment:
```suggestion
    then fail CI saying that semverver needs maintenance.
```

Reply:
The point about a minor version bump being an option should stay, but I like the suggestion of "needs maintenance" over "to be updated".